Source/Scene/Material.js

```diff
@@ -895,12 +915,16 @@ function createTexture2DUpdateFunction(uniformId) {
           promise = resource.fetchImage();
         }
 
-        when(promise, function (image) {
-          material._loadedImages.push({
-            id: uniformId,
-            image: image,
+        promise
+          .then(function (image) {
+            material._loadedImages.push({
+              id: uniformId,
+              image: image,
+            });
+          })
+          .otherwise(function () {
+            material._textures[uniformId] = material._defaultTexture;
```

Comment:
The last thing I noticed is that the texture isn't destroyed if it goes from "Cesium_Logo_Color.jpg" to "does_not_exist".

This is a bit hard to fix given the way the material code is currently arranged. The best approach is probably to destroy the existing texture in here, even though it happens outside the render loop.

The material code is due for a refactor but that's outside the scope of this PR.